genesyscloud/architect_schedules/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,262 @@
+package architect_schedules
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/leekchan/timeutil"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/constants"
+
+	"github.com/mypurecloud/platform-client-sdk-go/v129/platformclientv2"
+)
+
+func getAllArchitectSchedules(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getArchitectSchedulesProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	schedules, proxyResponse, getErr := proxy.getAllArchitectSchedules(ctx)
+	if getErr != nil {
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of schedule error: %s", getErr), proxyResponse)
+	}
+
+	for _, schedule := range *schedules {
+		resources[*schedule.Id] = &resourceExporter.ResourceMeta{Name: *schedule.Name}
+	}
+
+	return resources, nil
+}
+
+// createArchitectSchedulegs is used by the architect_schedulegs resource to create Genesys cloud architect schedules
+func createArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getArchitectSchedulesProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	divisionID := d.Get("division_id").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", start, err)
```

Comment:
I would use a BuildDiagError function here.  We never want to return a "naked" diagnostics error but instead make sure it is in an appropriate formate.